crates/biome_js_analyze/src/assists/correctness/organize_imports.rs

```diff
@@ -725,18 +637,22 @@ const NODE_BUILTINS: &[&str] = &[
 
 impl From<&str> for ImportCategory {
     fn from(value: &str) -> Self {
-        if value.starts_with("node:") || NODE_BUILTINS.contains(&value) {
-            Self::NodeBuiltin
-        } else if value.starts_with("npm:") {
-            Self::Npm
-        } else if value.starts_with("http:") || value.starts_with("https:") {
-            Self::Url
-        } else if value.starts_with('.') {
-            Self::Relative
-        } else if !value.contains(':') {
-            Self::Library
-        } else {
-            Self::Other
+        match value.chars().next().unwrap_or_default() {
+            '.' => Self::Relative,
+            _ => {
+                if let Some((protocol, _)) = value.split_once(':') {
+                    match protocol {
+                        "http" | "https" => Self::Url,
+                        "node" => Self::NodeBuiltin,
+                        "npm" => Self::Npm,
+                        _ => Self::Other,
+                    }
+                } else if NODE_BUILTINS.contains(&value) {
+                    Self::NodeBuiltin
+                } else {
+                    Self::Library
+                }
+            }
```

Comment:
IHMO the previous implementation was more readable, and I fail to see the benefit of this refactor.  Always, we create an iterator only to check the first character, I fail to see the benefit of this change too.

Reply:
I roll back to the previous style, except for the protocol part. I also use `binary_search` instead of `contains` for implicit Node built ins.

Reply:
Good call!